File: 2001-3000/2011_final_value_after_performing_operations/solution.py
from typing import List


class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        ops = {
            "++X": lambda x: x + 1,
            "X++": lambda x: x + 1,
            "--X": lambda x: x - 1,
            "X--": lambda x: x - 1,
        }

        value = 0

        for operation in operations:
            value = ops[operation](value)
        return value

File: 2001-3000/2011_final_value_after_performing_operations/solution.go
package main

import "fmt"

func finalValueAfterOperations(operations []string) int {
	var result int
	for i := 0; i < len(operations); i++ {
		switch operation := operations[i]; operation {
		case "++X":
			result++
		case "X++":
			result++
		case "--X":
			result--
		case "X--":
			result--
		}
	}
	return result
}

func main() {
	//operations := []string{"--X", "X++", "X++"}
	operations := []string{"++X", "++X", "X++"}
	fmt.Println(finalValueAfterOperations(operations))
}
